generator/filesystem.cpp

```diff
@@ -42,14 +52,24 @@ std::error_code canonicalize(const llvm::Twine &path, llvm::SmallVectorImpl<char
     std::string p = path.str();
```

Comment:
One can use `llvm::sys::fs::real_path` since clang 5.0

Reply:
Will it take care of the buffer size? Or do I still need to resize the buffer to path_max before?

Reply:
no need to resize, it resizes itself the buffer.
That's why you pass it a growing buffer rather than just a char*